packages/shopify-codemod/transforms/conditional-assign-to-if-statement.js

```diff
@@ -0,0 +1,62 @@
+function isConditionalAssignment(node) {
+  const isUsingOr = (node.operator === '||');
+  const isLeftCheckingVariable = (node.left.type === 'MemberExpression') || (node.left.type === 'Identifier');
+  const isRightAssignment = (node.right.type === 'AssignmentExpression');
+
+  return isUsingOr && isLeftCheckingVariable && isRightAssignment;
+}
+
+function isMemberEqual(aNode, bNode) {
+  return (
+    aNode.object.type === bNode.object.type &&
+    aNode.object.name === bNode.object.name &&
+    aNode.property.type === bNode.property.type &&
+    aNode.property.name === bNode.property.name
+  );
+}
+
+function isIdentifierEqual(aNode, bNode) {
+  return (
+    aNode.type === bNode.type &&
+    aNode.name === bNode.name
+  );
+}
+
+function isAssigningToConditionObject(node) {
+  const condition = node.left;
+  const assignment = node.right.left;
+  const isMemberExpression = (condition.type === 'MemberExpression');
+
+  return isMemberExpression
+    ? isMemberEqual(condition, assignment)
```

Comment:
Will this handle multiple levels of members? `foo.bar.baz || (foo.bar.baz = 'qux');`